docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,293 @@
+# HAP Rule Configuration
+
+> [!NOTE]
+> This feature is still being developed and will be available soon.
+
+ The Hyperscaler Account Pool (HAP) Rule Configuration is a functionality that allows to control pool selection from a configurable helm property. 
+ The HAP pool is a set of Gardener's SecretBindings K8S resources that share the same labels. 
+ The HAP Pool Selection is a process that for every SKR creation request decides on which secret binding is the most suitable to use. 
+ The process generates a list of [search labels](#search-labels) based on skr's physical attributes.
+ The [search labels](#search-labels) are used to query Gardener's Secret Bindings. 
+ [search labels](#search-labels) are generated by evaluating a Rule consisting of Rule Entries and if an entry applies to the request then the rule outputs its specific [search labels](#search-labels) to query the correct HAP Pool.
+ More about HAP process can be found in [Hyperscaler Account Pool](03-10-hyperscaler-account-pool.md) document.
+
+![Rules Evaluation Process](../assets/hap-rules-general.drawio.svg)
+
+## Rule Configuration
+
+The Rule is configured in values.yaml in `hap.rule` property.
+It is a list of strings, each of which is a single Rule Entry that corresponds to a single HAP Pool. 
+Each Rule Entry is evaluated during a cluster provisioning process.
+If more than one rule applies the best matching rule is selected based on the rule [priority](#priority). The following examples presents the Rule Configuration in values.yaml:
+
+```
+hap: 
+  rule: 
+  - rule_1
+  - rule_2
+  ...
+  - rule_n
+```
+
+Every HAP Pool that the Rule Entry represents must be preconfigured in a Gardener Cluster that KEB connects to.
+
+## Rule Format 
+
+The following example shows the Rule Entry format:
+
+```
+PLAN(INPUT_ATTR_1=VAL_1, INPUT_ATTR_2=VAL_2, ..., INPUT_ATTR_N=VAL_N) -> OUTPUT_ATTR_1, OUTPUT_ATTR_2, ..., OUTPUT_ATTR_N
```

Comment:
This suggests that there are N pairs (attribute=value) and N output attributes which not necessarily is true.